### PR TITLE
change DYNAMIC_THREAD_COUNT to be an AtomicUsize

### PR DESCRIPTION
**Pull request by kolapapa**
https://github.com/async-rs/async-std/pull/141

### CODE REVIEW

File: src/task/task_id.rs [hunk 1]
```diff
@@ -13,15 +13,15 @@ use std::sync::atomic::{AtomicU64, Ordering};
...
 /// ```
 #[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
-pub struct TaskId(pub(crate) u64);
+pub struct TaskId(pub(crate) usize);
 
 impl TaskId {
     /// Generates a new `TaskId`.
     pub(crate) fn generate() -> TaskId {
-        static COUNTER: AtomicU64 = AtomicU64::new(1);
+        static COUNTER: AtomicUsize = AtomicUsize::new(1);
```
Comment (stjepang) [inline src/task/task_id.rs]: This should still be an `AtomicU64`, I'm afraid. What we can do is use `AtomicCell<u64>` from `crossbeam-utils`.

Comment (kolapapa) [reply src/task/task_id.rs]: I have changed it as you suggest

**Comment by kolapapa** [comment]
https://github.com/async-rs/async-std/pull/286  Is this PR ready to merge? Then I'll close this~It looks better

**Comment by yoshuawuyts** [comment]
@kolapapa it seems this PR has merge conflicts with master. Could this perhaps be rebased? Thanks!

**Pull request closed without merging**